src/commands/run.rs

```diff
@@ -209,16 +211,49 @@ pub async fn run(args: Args) -> Result<()> {
         .run()
         .await
     {
-        Ok(results) => results,
+        Ok(results) => results.into_inner(),
         Err(errors) => {
             // SAFETY: this is a non-empty, so it must always have a first
             // element.
             bail!(errors.into_iter().next().unwrap())
         }
     };
 
+    // Emits diagnostics for all analyzed documents
+    let mut errors = 0;
+    for result in &results {
+        let diagnostics = result.document().diagnostics();
+        if !diagnostics.is_empty() {
+            let path = result.document().path().to_string();
+            let source = result.document().root().text().to_string();
+
+            errors += diagnostics
+                .iter()
+                .filter(|d| d.severity() == Severity::Error)
```

Comment:
we frequently do this, and I wonder if we should have a "shortcut". OOS for this PR, just a thought.